lib/Reconciler.lua

```diff
@@ -64,75 +56,47 @@ local Reconciler = {}
 
 Reconciler._singleEventManager = SingleEventManager.new()
 
---[[
-	Is this element backed by a Roblox instance directly?
-]]
-local function isPrimitiveElement(element)
-	if type(element) ~= "table" then
-		return false
-	end
-
-	return type(element.component) == "string"
-end
-
---[[
-	Is this element defined by a pure function?
-]]
-local function isFunctionalElement(element)
-	if type(element) ~= "table" then
-		return false
-	end
-
-	return type(element.component) == "function"
-end
-
---[[
-	Is this element defined by a component class?
-]]
-local function isStatefulElement(element)
-	if type(element) ~= "table" then
-		return false
-	end
-
-	return type(element.component) == "table"
-end
-
 --[[
 	Destroy the given Roact instance, all of its descendants, and associated
 	Roblox instances owned by the components.
 ]]
 function Reconciler.unmount(instanceHandle)
 	local element = instanceHandle._element
 
-	if isPrimitiveElement(element) then
-		-- We're destroying a Roblox Instance-based object
+	if type(element) == "table" then
```

Comment:
Instead of indenting the entire function one level, we should probably just early-out with an error if `element` isn't a table.

That being said, in this function, `element` should _always_ be a table, so this check is just redundant.